azext_edge/edge/providers/check/base/pod.py

```diff
@@ -28,228 +28,186 @@ def decorate_pod_phase(phase: str) -> Tuple[str, str]:
-            detail_level=detail_level,
+            display=Padding(table, (0, 0, 0, padding)),
         )
 
-        check_manager.add_target_eval(
+    if add_footer:
+        footer = ":magnifying_glass_tilted_left:" + colorize_string(
+            " See more details by attaching : --detail-level 1 or --detail-level 2"
+        )
+        check_manager.add_display(
             target_name=target,
-            status=pod_eval_status,
-            value=pod_eval_value,
             namespace=namespace,
-            resource_name=target_service_pod,
+            display=Padding(footer, (0, 0, 0, padding)),
         )
 
 
-def _add_pod_health_display(
+def process_pod_status(
     check_manager: CheckManager,
     target: str,
+    pod: V1Pod,
     namespace: str,
-    pod_name: str,
-    pod_phase_deco: str,
-    pod_conditions: List,
-    pod_eval_status: str,
-    conditions_readiness: bool,
-    conditions_display_list: List[Tuple[str, str]],
-    unknown_conditions_display_list: List[Tuple[str, str]],
-    display_padding: int,
     detail_level: int = ResourceOutputDetailLevel.summary.value,
-) -> None:
-    pod_health_status = "[green]Healthy[/green]"
-
-    if pod_eval_status == CheckTaskStatus.error.value:
-        pod_health_status = "[red]Unhealthy[/red]"
-    elif pod_eval_status == CheckTaskStatus.warning.value:
-        pod_health_status = "[yellow]Indeterminate[/yellow]"
-
-    pod_health_text = f"Pod {{[bright_blue]{pod_name}[/bright_blue]}} is {pod_health_status}"
+) -> Tuple[List[str], str]:
 
-    if detail_level != ResourceOutputDetailLevel.summary.value:
-        pod_health_text = f"\n{pod_health_text}"
+    def _decorate_pod_condition(condition: bool) -> Tuple[str, str]:
+        if condition:
+            return f"[green]{condition}[/green]", CheckTaskStatus.success.value
+        return f"[red]{condition}[/red]", CheckTaskStatus.error.value
 
-    check_manager.add_display(
+    target_service_pod = f"pod/{pod.metadata.name}"
+
+    pod_conditions = [
+        f"{target_service_pod}.status.phase",
+        f"{target_service_pod}.status.conditions",
+    ]
+
+    if check_manager.targets.get(target, {}).get(namespace, {}).get("conditions", None):
+        check_manager.add_target_conditions(target_name=target, namespace=namespace, conditions=pod_conditions)
+    else:
+        check_manager.set_target_conditions(target_name=target, namespace=namespace, conditions=pod_conditions)
+
+    pod_dict = pod.to_dict()
+    pod_name = pod_dict["metadata"]["name"]
+    pod_phase = pod_dict.get("status", {}).get("phase")
+    pod_conditions: list = pod_dict.get("status", {}).get("conditions", [])
+    pod_phase_deco, status = decorate_pod_phase(pod_phase)
+
+    pod_eval_value = {}
+    pod_eval_status = status
+    pod_eval_value["status.phase"] = pod_phase
+
+    conditions_readiness = True
+    conditions_display_list: List[Tuple[str, str]] = []
+    unknown_conditions_display_list: List[Tuple[str, str]] = []
```

Comment:
It looks like these are lists of tuples, like `("key: value", formatted)`, but then later on they are joined together into one string.

Could we just create them as one string when they're added to this list?

Something like:
```py
lines = [
    f"{condition_type}: {pod_condition_deco}",
    formatted_reason
]
conditions_display_list.append('/n'.join(lines))
```

Reply:
I separated them because we are not showing the formatted_reason in all scenarios, for detail level we will only show reason when one of the condition failed, but in verbose we will always show when there is one. Will think if there is a better way to do this

Reply:
for this one, I also made a namedTuple called PodStatusConditionResult, hopefully that can be more cleared!